consensus/mint-client/src/config.rs

```diff
@@ -2,14 +2,143 @@
 use mc_account_keys::PublicAddress;
 use mc_api::printable::PrintableWrapper;
-use mc_crypto_keys::{DistinguishedEncoding, Ed25519Private};
-use mc_transaction_core::mint::constants::NONCE_LENGTH;
+use mc_crypto_keys::{DistinguishedEncoding, Ed25519Pair, Ed25519Private, Ed25519Public, Signer};
+use mc_crypto_multisig::{MultiSig, SignerSet};
+use mc_transaction_core::mint::{
+    constants::NONCE_LENGTH, MintConfig, MintConfigTx, MintConfigTxPrefix, MintTx, MintTxPrefix,
+};
 use mc_util_uri::ConsensusClientUri;
-use std::{convert::TryFrom, fs};
+use rand::{thread_rng, RngCore};
+use std::{convert::TryFrom, fs, path::PathBuf};
+
+#[derive(Args)]
+pub struct MintConfigTxParams {
+    /// The key(s) to sign the transaction with.
+    #[clap(long = "signing-key", required =true, use_value_delimiter = true, parse(try_from_str = load_key_from_pem), env = "MC_MINTING_SIGNING_KEYS")]
+    signing_keys: Vec<Ed25519Private>,
+
+    /// The token id we are minting.
+    #[clap(long, env = "MC_MINTING_TOKEN_ID")]
+    token_id: u32,
+
+    /// Tombstone block.
+    #[clap(long, env = "MC_MINTING_TOMBSTONE")]
+    tombstone: Option<u64>,
+
+    /// Nonce.
+    #[clap(long, parse(try_from_str = FromHex::from_hex), env = "MC_MINTING_NONCE")]
+    nonce: Option<[u8; NONCE_LENGTH]>,
+
+    /// Mint configs. Each configuration must be of the format: <mint
+    /// limit>:<signing threshold>:<signer 1 public keyfile>[:<signer 2
+    /// public keyfile....>]. For example:
+    /// 10000:2:signer1.pem:signer2.pem:signer3.pem defines a minting
+    /// configuration capable of minting up to 1000 tokens: and requiring 2
+    /// out of 3 signers.
+    #[clap(long = "config", parse(try_from_str = parse_mint_config), env = "MC_MINTING_CONFIGS")]
+    // Tuple of (mint limit, SignerSet)
+    configs: Vec<(u64, SignerSet<Ed25519Public>)>,
+}
+
+impl MintConfigTxParams {
+    pub fn try_into_mint_config_tx(
+        self,
+        fallback_tombstone_block: impl Fn() -> u64,
+    ) -> Result<MintConfigTx, String> {
+        let tombstone_block = self.tombstone.unwrap_or_else(fallback_tombstone_block);
+        let nonce = get_or_generate_nonce(self.nonce);
+        let token_id = self.token_id;
+        let prefix = MintConfigTxPrefix {
+            token_id,
+            configs: self
+                .configs
+                .into_iter()
+                .map(|(mint_limit, signer_set)| MintConfig {
+                    token_id,
+                    mint_limit,
+                    signer_set,
+                })
+                .collect(),
+            nonce,
+            tombstone_block,
+        };
+
+        let message = prefix.hash();
+        let signature = MultiSig::new(
+            self.signing_keys
+                .into_iter()
+                .map(|signer| {
+                    Ed25519Pair::from(signer)
+                        .try_sign(message.as_ref())
+                        .unwrap()
+                })
+                .collect(),
```

Comment:
Let's return an error here?
```suggestion
                        .map_err(|e| format!("Failed to sign MintConfigTxPrefix: {}", e))?
                })
                .collect::<Result<Vec<_>, _>()?,
```